csrc/id_model/id_graphs.cpp

```diff
@@ -1762,9 +1762,73 @@ std::unordered_map<IdGroup, IterDomain*> IterDomainGraphs::
     }
   }
 
-  std::cerr << "Loop promotion map:\n";
+  // TODO: cleanup
```

Comment:
At this point, we have a map from IEL groups to promotion domains. Here, the code below just populates `loop_promotion_map_` with the IEL map. Some loop groups don't have any domain in the IEL map. In that case, `loop_graph_copy_promotion_map` should have the correct promotion.